webapp/store/models.py

```diff
@@ -0,0 +1,145 @@
+
+
+cs = CharmStore("https://api.jujucharms.com/v5")
+
+
+def get_charm_or_bundle(name):
+    try:
+        entity_data = cs.entity(name, True)
+    except EntityNotFound:
+        return abort(404, "Entity not found {}".format(name))
+
+    charm_or_bundle_id = entity_data['Id']
+    ref = references.Reference.from_string(charm_or_bundle_id)
+    meta = entity_data.get('Meta', None)
+    bzr_url, revisions = _extract_from_extrainfo(meta, ref)
+    bugs_url, homepage = _extract_from_commoninfo(meta)
+    revision_list = meta['revision-info'].get('Revisions')
+    latest_revision = {
+        'id': int(revision_list[0].split('-')[-1]),
+        'full_id': revision_list[0],
+        'url': '{}/{}'.format(
+            meta['charm-metadata']['Name'],
+            int(revision_list[0].split('-')[-1])
+        )}
+
+    charm_or_bundle = {
+        'archive_url': cs.archive_url(ref),
+        'bugs_url': bugs_url,
+        'bzr_url': bzr_url,
+        'entity_data': entity_data,
+        'files': _get_entity_files(ref, meta['manifest']),
+        'homepage': homepage,
+        'icon': cs.charm_icon_url(charm_or_bundle_id),
+        'id': charm_or_bundle_id,
+        'card_id': ref.path(),
+        'latest_revision': latest_revision,
+        'options': meta.get('charm-config', {}).get('Options'),
+        'owner': entity_data.get('owner', {}).get('User'),
+        'provides': meta['charm-metadata'].get('Provides'),
+        'requires': meta['charm-metadata'].get('Requires'),
+        'resources': _extract_resources(ref, meta.get('resources', {})),
+        'readme': _render_markdown(
+            cs.entity_readme_content(charm_or_bundle_id)
+        ),
+        'revision_list': revision_list,
+        'revision_number': ref.revision,
+        'revisions': revisions,
+        'series': meta.get('supported-series', {}).get('SupportedSeries'),
+        'is_charm': 'charm-metadata' in meta
+    }
+
+    return charm_or_bundle
+
+
+def _get_entity_files(ref, manifest=None):
+    try:
+        files = cs.files(ref, manifest=manifest) or {}
+    except EntityNotFound:
+        files = {}
+    return files
+
+
+def _extract_resources(ref, resources):
+    """Extract data from resources metadata.
+    @param ref The reference of the entity.
+    @param resources the resources metadata associated with the entity.
+    @return a dictionary of resource name and an array containing
+            the file extension, the file link of the resource.
+    """
+    result = {}
+    for resource in resources:
+        resource_url = ""
+        if resource['Revision'] >= 0:
+            resource_url = cs.resource_url(
+                ref,
+                resource['Name'],
+                resource['Revision']
+            )
+        result[resource['Name']] = [
+            os.path.splitext(resource['Path'])[1],
+            resource_url
+        ]
+
+    return result
+
+
+def _get_bug_url(name, bugs_url):
+    '''Create a link to the bug tracker on Launchpad.
```

Comment:
Se below

Reply:
and use `"` instead of `'`

Reply:
Unless Flake8 picks up on this, I'm not sure it's something we need to be anal about.